### PR TITLE
http request plugin implementation

### PR DESCRIPTION
**Pull request by senthilhns**
http request plugin implementation

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (Ompragash) [inline README.txt]: let's remove this file and only keep README.md file

File: README.md [hunk 2]
```diff
@@ -1 +1,52 @@
...
+
+```text
+docker run --rm -e PLUGIN_PARAM1=foo -e PLUGIN_PARAM2=bar \
+  -e DRONE_COMMIT_SHA=8f51ad7884c5eb69c11d260a31da7a745e6b78e2 \
+  -e DRONE_COMMIT_BRANCH=master \
+  -e DRONE_BUILD_NUMBER=43 \
+  -e DRONE_BUILD_STATUS=success \
+  -w /drone/src \
+  -v $(pwd):/drone/src \
+  senthilhns/drone_http_request_plugin
```
Comment (Ompragash) [inline README.md]: update the example with inputs supported by httpRequest plugin

File: README.md [hunk 3]
```diff
@@ -1 +1,52 @@
...
+Build the plugin binary:
+
+```text
+scripts/build.sh
+```
+
+Build the plugin image:
+
+```text
+docker build -t senthilhns/drone_http_request_plugin -f docker/Dockerfile .
```
Comment (Ompragash) [inline README.md]:
```suggestion
docker build -t local/drone_http_request_plugin -f docker/Dockerfile .
```

File: README.md [hunk 4]
```diff
@@ -1 +1,52 @@
...
+name: default
+
+steps:
+- name: run senthilhns/drone_http_request_plugin plugin
+  image: senthilhns/drone_http_request_plugin
+  pull: if-not-exists
+  settings:
+    param1: foo
+    param2: bar
+```
```
Comment (Ompragash) [inline README.md]: remove this section

File: plugin/plugin.go [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+		p.HttpRequestCancelContext = nil
+	}
+
+	if p.BodyIoReader != nil {
+		if closer, ok := p.BodyIoReader.(io.Closer); ok {
+			err = closer.Close()
+		}
+	}
+
+	LogPrintln(p, "DeInit() called")
```
Comment (Ompragash) [inline plugin/plugin.go]:
```suggestion
	if err != nil {
		LogPrintln(p, "DeInit() called with error:", err)
	}
```

let's log this only when there's an error

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (senthilhns) [reply README.md]: updated

File: README.md (hunk 4 above)
Comment (senthilhns) [reply README.md]: removed

File: README.md [hunk 6]
```diff
@@ -1 +1,68 @@
...
+
+# Usage
+
+The following settings changes this plugin's behavior.
+
+* param1 (optional) does something.
+* param2 (optional) does something different.
+
+Below is an e
+
```
Comment (Ompragash) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -1 +1,68 @@
...
+Build the plugin binary:
+
+```text
+scripts/build.sh
+```
+
+Build the plugin image:
+
+```text
+docker build -t local/drone_http_request_plugin -f docker/Dockerfile .
```
Comment (Ompragash) [inline README.md]:
```suggestion
docker build -t local/httprequest -f docker/Dockerfile .
```

File: README.md [hunk 8]
```diff
@@ -1 +1,68 @@
...
+Execute the plugin from your current working directory:
+
+```text
+Simple GET request
+docker run --rm \
+-e PLUGIN_URL='https://httpbin.org/get' \
+-e PLUGIN_HTTP_METHOD='GET' \
+-e PLUGIN_HEADERS='Content-Type:application/json' \
+-e PLUGIN_TIMEOUT='30' \
+  -w /drone/src  -v $(pwd):/drone/src local/drone_http_request_plugin
```
Comment (Ompragash) [inline README.md]:
```suggestion
  -w /drone/src  -v $(pwd):/drone/src local/httprequest
```

File: README.md [hunk 9]
```diff
@@ -1 +1,68 @@
...
+
+
+GET request with response logged in to a file
+docker run --rm \
+-e PLUGIN_URL='https://httpbin.org/get' \
+-e PLUGIN_HTTP_METHOD='GET' \
+-e PLUGIN_HEADERS='Content-Type:application/json' \
+-e PLUGIN_TIMEOUT='30' \
+-e PLUGIN_OUTPUT_FILE='/tmp/output_6008.txt' \
+  -w /drone/src  -v $(pwd):/drone/src local/drone_http_request_plugin
```
Comment (Ompragash) [inline README.md]:
```suggestion
  -w /drone/src  -v $(pwd):/drone/src local/httprequest
```

File: README.md [hunk 10]
```diff
@@ -1 +1,68 @@
...
+
+GET request with proxy usage
+docker run --rm \
+-e PLUGIN_URL='https://httpbin.org/get' \
+-e PLUGIN_HTTP_METHOD='GET' \
+-e PLUGIN_HEADERS='Content-Type:application/json' \
+-e PLUGIN_TIMEOUT='30' \
+-e PLUGIN_IGNORE_SSL='true' \
+-e PLUGIN_PROXY='http://localhost:8888' \
+  -w /drone/src  -v $(pwd):/drone/src local/drone_http_request_plugin
```
Comment (Ompragash) [inline README.md]:
```suggestion
  -w /drone/src  -v $(pwd):/drone/src local/httprequest
```

File: README.md [hunk 11]
```diff
@@ -1 +1,68 @@
...
+
+Simple POST request
+docker run --rm \
+-e PLUGIN_URL='https://httpbin.org/post' \
+-e PLUGIN_HTTP_METHOD='POST' \
+-e PLUGIN_HEADERS='Content-Type:application/json' \
+-e PLUGIN_REQUEST_BODY='{"name":"drone"}' \
+-e PLUGIN_TIMEOUT='30' \
+  -w /drone/src  -v $(pwd):/drone/src local/drone_http_request_plugin
```
Comment (Ompragash) [inline README.md]:
```suggestion
  -w /drone/src  -v $(pwd):/drone/src local/httprequest
```

File: docker/manifest.tmpl [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    platform:
+      variant: v7
+      architecture: arm
+      os: linux
+  -
+    image: senthilhns/drone_http_request_plugin:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
+    platform:
+      variant: v6
+      architecture: arm
+      os: linux
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
image: plugins/httprequest:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}latest{{/if}}
{{#if build.tags}}
tags:
{{#each build.tags}}
  - {{this}}
{{/each}}
{{/if}}
manifests:
  -
    image: plugins/httprequest:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-amd64
    platform:
      architecture: amd64
      os: linux
  -
    image: plugins/httprequest:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm64
    platform:
      variant: v8
      architecture: arm64
      os: linux
```

File: scripts/build.sh [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+# disable cgo
+export CGO_ENABLED=0
+
+set -e
+set -x
+
+# linux
+GOOS=linux GOARCH=amd64 go build -o release/linux/amd64/plugin
+#GOOS=linux GOARCH=arm64 go build -o release/linux/arm64/plugin
```
Comment (Ompragash) [inline scripts/build.sh]:
```suggestion
GOOS=linux GOARCH=arm64 go build -o release/linux/arm64/plugin
```

**Review by Ompragash — APPROVED** [review]

**Pull request merged**